### DISCUSSION TITLE
How to OUTRIGHT NOT TRIGGER a deployment unless on production branch?

### DISCUSSION THREAD

**Discussion by TomDo1234** [vercel/community#2890] [Help] [answered]
### Summary

You can use ignore build step... But it still triggers a deployment which counts towards 100 a day limit

### Steps to Reproduce

Use an ignore build step and observe it count towards 100 a day limit.

I have 3 sites that are variants of each other, so I have 3 branches on 1 git repository.

I can only deploy 11 times a day.

this is because I have 1 project for each, so 3 deployments per update to all 3 sites. But then the other branches also "count" as builds for all 3 projects so that is an additional 6

6 + 3 = 9. 9 deployments per update!

**Comment by smaeda-ks** [comment, marked as answer]
@TomDo1234 possible workaround is to disable the GitHub integration and trigger deployment via CLI using CI/CD workflow:
https://vercel.com/guides/how-can-i-use-github-actions-with-vercel

Have you considered that?

**Reply by TomDo1234** [reply]
In the end I did consider this... But was too lazy to implement it. I guess I will have to do it.

**Reply by smaeda-ks** [reply]
you can also configure `git.deploymentEnabled` in your `vercel.json` file:
https://vercel.com/docs/concepts/projects/project-configuration/git-configuration#git.deploymentenabled

**Reply by TomDo1234** [reply]
That doesn't work it seems.

I have 3 branches, for the sake of confidentiality I will name A , B and C .

in A I make vercel json git.deploymentenabled false for everything except A

in B I do the same but for B, and for C the same as well.

the 3 branches still trigger builds in the other 2 projects!

I guess I will need to use CI/CD workflow.

**Comment by essenmitsosse** [comment]
I find this answer rather unsatisfying. I don't need any deployments on development branches at all. Yet, I constantly get a big red ❌ for every commit and PR, because Vercel fails (without any proper error message by the way), just because I can't ignore a branch and every single commit still counts towards the limit. This is just insanely unnecessary.
